src/Rules/ThrowsPhpDocRule.php

```diff
@@ -140,6 +149,12 @@ public function getNodeType(): string
 	 */
 	public function processNode(Node $node, Scope $scope): array
 	{
+		$method = $scope->getFunction();
+
+		if ($method instanceof MethodReflection && $this->isWhitelistedMethod($method)) {
```

Comment:
This part of the code is executed very often - every type of node inside a method, right?

So please add `$node instanceof FunctionEnd` to the condition.